src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -119,7 +120,11 @@ public HttpConnectionPool(HttpConnectionPoolManager poolManager, HttpConnectionK
             }
 
             _http2Enabled = _poolManager.Settings._maxHttpVersion >= HttpVersion.Version20;
-            _http3Enabled = _poolManager.Settings._maxHttpVersion >= HttpVersion.Version30 && (_poolManager.Settings._quicImplementationProvider ?? QuicImplementationProviders.Default).IsSupported;
+            // TODO: Replace with Platform-Guard Assertion Annotations once https://github.com/dotnet/runtime/issues/44922 is finished
+            if (OperatingSystem.IsLinux() || OperatingSystem.IsWindows() || OperatingSystem.IsOSPlatform("osx"))
```

Comment:
Here you use OperatingSystem.IsOSPlatform("osx") and bellow OperatingSystem.IsMacOS() it would be nice to use same one.

Reply:
Fixed.